events/abc/client.go

```diff
@@ -0,0 +1,137 @@
+package abc
+
+import (
+	"github.com/checkout/checkout-sdk-go/client"
+	"github.com/checkout/checkout-sdk-go/common"
+	"github.com/checkout/checkout-sdk-go/configuration"
+)
+
+type Client struct {
+	configuration *configuration.Configuration
+	apiClient     client.HttpClient
+}
+
+func NewClient(configuration *configuration.Configuration, apiClient client.HttpClient) *Client {
+	return &Client{
+		configuration: configuration,
+		apiClient:     apiClient,
+	}
+}
+
+func (c *Client) RetrieveAllEventTypes(
+	query QueryRetrieveAllEventType,
+) (*EventTypesResponse, error) {
+	auth, err := c.configuration.Credentials.GetAuthorization(configuration.SecretKey)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := common.BuildQueryPath(eventTypes, query)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var response EventTypesResponse
+	err = c.apiClient.Get(url, auth, &response)
+	if err != nil {
+		return nil, err
+	}
+
+	return &response, nil
+}
+
+func (c *Client) RetrieveEvents(query QueryRetrieveEvents) (*EventsPageResponse, error) {
```

Comment:
`QueryRetrieveEvents` seems odd and doesn't make sense, it is probably better `QueryEvents` or `RetrieveEventsRequest`

Reply:
OK, thank you.